src/ai/defaultai.cc

```diff
@@ -1710,7 +1729,8 @@ void DefaultAI::update_buildable_field(BuildableField& field) {
 						continue;
 					}
 				}
-				uint8_t amount = map.find_immovables(
+				uint8_t amount = 0;
+				amount = map.find_immovables(
```

Comment:
What's the purpose of splitting declaration and assignment here?

Reply:
Well IIRC this is a leftover of the attempts to fix the bug that quarries are build even when rocks are already depleted. In the first instance I beleived that amount might have not been defined if no immovables are found, but the fix was done some lines above. As I checked that find_immovables will return 0 if no immovables found I'll revert this